thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/views/CondensedAnomalyTimelinesView.java

```diff
@@ -182,9 +183,33 @@ public static CondensedAnomalyTimelinesView fromAnomalyTimelinesView(AnomalyTime
    * @return a compressed CondensedAnomalyTimelinesView
    */
   public CondensedAnomalyTimelinesView compress() {
+    if (timeStamps.size() == 0) {
+      return this;
+    }
+    try {
+      if (this.toJsonString().length() > DEFAULT_MAX_LENGTH) {
+        // First try rounding up
+        roundUp();
```

Comment:
Good suggestion. That also avoids some code duplications. At first I wanted to avoid putting it in compress(DEFAULT_MAX_LENGTH) since that one is called recursively. But now I think it is actually more robust to have this roundup every time we do compression. modified.